crypto/keyring/keyring.go

```diff
@@ -217,95 +242,81 @@ func newKeystore(kr keyring.Keyring, opts ...Option) keystore {
 		optionFn(&options)
 	}
 
-	return keystore{kr, options}
+	return keystore{kr, cdc, options}
 }
 
 func (ks keystore) ExportPubKeyArmor(uid string) (string, error) {
-	bz, err := ks.Key(uid)
+	k, err := ks.Key(uid)
 	if err != nil {
 		return "", err
 	}
 
-	if bz == nil {
-		return "", fmt.Errorf("no key to export with name: %s", uid)
+	key, err := k.GetPubKey()
+	if err != nil {
+		return "", err
 	}
 
-	return crypto.ArmorPubKeyBytes(legacy.Cdc.MustMarshal(bz.GetPubKey()), string(bz.GetAlgo())), nil
+	return crypto.ArmorPubKeyBytes(legacy.Cdc.MustMarshal(key), key.Type()), nil
 }
 
 func (ks keystore) ExportPubKeyArmorByAddress(address sdk.Address) (string, error) {
-	info, err := ks.KeyByAddress(address)
+	k, err := ks.KeyByAddress(address)
 	if err != nil {
 		return "", err
 	}
 
-	return ks.ExportPubKeyArmor(info.GetName())
+	return ks.ExportPubKeyArmor(k.Name)
 }
 
+// we use ExportPrivateKeyFromLegacyInfo(info LegacyInfo) (cryptotypes.PrivKey, error) { for LegacyInfo
 func (ks keystore) ExportPrivKeyArmor(uid, encryptPassphrase string) (armor string, err error) {
-	priv, err := ks.ExportPrivateKeyObject(uid)
+	fmt.Println("ExportPrivKeyArmor start")
+	_, priv, err := ks.ExportPrivateKeyObject(uid)
 	if err != nil {
 		return "", err
 	}
+	fmt.Println("ExportPrivKeyArmor done")
 
-	info, err := ks.Key(uid)
-	if err != nil {
-		return "", err
-	}
-
-	return crypto.EncryptArmorPrivKey(priv, encryptPassphrase, string(info.GetAlgo())), nil
+	return crypto.EncryptArmorPrivKey(priv, encryptPassphrase, priv.Type()), nil
 }
 
 // ExportPrivateKeyObject exports an armored private key object.
-func (ks keystore) ExportPrivateKeyObject(uid string) (types.PrivKey, error) {
-	info, err := ks.Key(uid)
+func (ks keystore) ExportPrivateKeyObject(uid string) (*Record, types.PrivKey, error) {
```

Comment:
Is there a reason why we changed the function signature? I would keep the old one

Reply:
yeah it does make sense!